src/lib.rs

```diff
@@ -235,6 +242,19 @@ impl Receiver {
         })
     }
 
+    /// Attempt to read a message without blocking. Returns a slice directly from the ring buffer
+    ///
+    /// This will return `Ok(None)` if there are no messages immediately available.
+    pub fn try_recv_direct<'a>(&'a self) -> Result<Option<&'a [u8]>>{
+        Ok(if let Some((value, position)) = self.try_recv_direct_0()? {
+            self.seek(position)?;
```

Comment:
This is both unsafe and unsound.  The call to `seek` tells the writer(s) that we're done reading the buffer, which is not true, since we're about to return a shared reference to the caller.  That violates Rust's aliasing rules because any writer(s) will believe they have unique, mutable access to something the caller has shared, immutable access to.

`ZeroCopyContext` exists precisely to avoid this problem.

Reply:
that's a valid point, I'll drop that method for the `direct` version, it indeed should only every be called within the usage of `ZeroCopyContext`